pkg/server/server.go

```diff
@@ -20,8 +21,33 @@ func handler (w http.ResponseWriter, r *http.Request){
 	fmt.Fprintf(w, "<h1>books</h1>%s", content);
 }
 
+func featured(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var titles = [4]string{"The jungle", "The test", "bazinga", "shelldon"}
+	var all_docs []interface{}
+
+	for _, title := range titles {
+		//TODO: better API???
+		query, err := booksapi.QueryTitle(title, 1)
+		if err != nil {
+			//TODO: error type etc
+			w.WriteHeader(http.StatusInternalServerError)
+			error_response := map[string]string{
```

Comment:
Bit of a spitball

```go
error_response := struct {
    ErrorMsg string `json:"error"`
}{ErrorMsg : err.Error()}
```

Reply:
fair, that seem bit more helpful yes haha